[4.0] Simplify some installation lang strings

C-Lodder opened the pull request:
### Summary of Changes

When installing Joomla 4, I believe the strings for inputs are too long.

For example, when entering the username, email and password for your account, the header for this page is `Login Data`. I automatically know I need to add my login details for the site.

Therefore for the username field, we don't need a long string like:

> Set the username for your Super User account.

It can just be:

> Username

Easier to read and gets straight to the point.

![screeny](https://user-images.githubusercontent.com/2019801/33146124-f72ed7d4-cfba-11e7-8cc3-70d099713b6e.png)


There are other strings that can be done, but this PR just addresses views 1 and 2.

Note, also replaced the `heart` icon with a `user` icon.

C-Lodder committed:
simplify some installation lang strings

dgrammatiko commented:
This is something for the a11y so calling @zwiastunsw and the english language team so calling also @brianteeman. I guess these two teams need to cooperate for the final strings, but I don't think a string like `Username` is accessible or descriptive enough for screen readers (or at least this is what's in my mind from the London sprint, might be wrong thou)

C-Lodder commented:
`Account username` could work too. I just don't see the need for a whole line to define what an input field is for

dgrammatiko commented:
A setence needs a verb, I guess

wilsonge commented:
For people relatively new to cms’ we should make it clear this is a master account. I don’t care if that’s a fieldset description or in the field labels

brianteeman commented:
As far as I know there is no en-gb team anymore -at least for the last 362 days

brianteeman commented:
A quick look at the strings and I agree with @wilsonge these changes create confusion not simplify thimgs

C-Lodder commented:
Speaking as a power user, before I was able to fly through the installation as I could quickly identify a key word in the shorter phrases, so I knew what to input and where to input it. Call me impatient, but now it's like having to carefully follow a recipe.

Furthermore, why are we using different words to start a sentence (`Enter`, `Set`, `Specify`, `Either`)?

![untitled](https://user-images.githubusercontent.com/2019801/33155860-1865cc1a-cfec-11e7-84e9-65deb34a3ab4.png)

dgrammatiko commented:
@C-Lodder how about placeholders like `Username` etc. Then a power user will not even have to look at the left part (label)?

C-Lodder commented:
@dgt41 Nah cause it's adding even more clutter to which **should** be more simplistic. Personally I've always been a fan of form helpers, as in smaller text below the input as a brief description if needed.

Example of that is the "Select database type" in the screenshot above, (although not required for that field)

brianteeman commented:
**Username** - Either one you created or supplied by your host
**Password** - Either one you created or supplied by your host
**Database Name** - Either one you created or supplied by your host
**Table Prefix** - The generated one below or create your own

??

brianteeman commented:
Note it is NOT Database **Table** Name

dgrammatiko commented:
> Username - Either one you created or supplied by your host

are these the labels you suggest or a combo label placeholder?

brianteeman commented:
Either as a label as is
Or with the bold as a label and the rest as a description

placeholders are evil!!!

coolcat-creations commented:
For Login Data: Can we put the information in the headline like
"Create Joomla Super Administrator Account" or similar ?
Then the labels Username, Password and Your E-Mail address would be enough.

For Database: It can be also one i did neither created nor is supplied by my host, when i create a database local i can create it with the installer itself. So what about:
Headline: "Specify the Database to connect Joomla to"  ?

And just at table prefix the further information "generated or specify your own"

zwiastunsw commented:
My proposal:
1. Instead *Login data*: **Create Super User account**.  Then labels are enough: Username, Email address, Password.
2. Instead *Database Configuration*: **Configure Database**. And then as suggested by @brianteeman: 
* **Database type** - Select from the list below
* **Hostname** - Either "localhost" or a name provided by your host
* **Database Username** - Either one you created or supplied by your host
* **Database Password** - Either one you created or supplied by your host
* **Database Name** - Either one you created or supplied by your host
* **Table prefix** - The generated one below or create your own

@dgt41 : Placeholders are wrong, because they are not read by screen readers.

coolcat-creations commented:
Can we mention "Either one you created or supplied by your host" only once? Is it even necessary?

mbabker commented:
It should be said once to guide users, definitely doesn't need to be repetitive though. A short sentence at the top of the block should be enough.

dgrammatiko commented:
>  Is it even necessary?

Well, turn on voice over, switch off your monitor and reload the page and tab through...

zwiastunsw commented:
Tests with users would be needed here. This is a problem of usability rather than a11y. However, such a solution seems to provide a complete instruction for the visually impaired and as well as the visionaries. This is in accordance with the principle: "Don't make me think" (Steve Krug).

alikon commented:
this is one of the few point where you should think imho

coolcat-creations commented:
So what other possibility is there for a Database if not one I created or someone gave to me.

mbabker commented:
https://codex.wordpress.org/images/thumb/5/5a/install-step3.png/600px-install-step3.png

Short sentence at the top of the box explaining we save stuff in a database and this info is either something you set up or were given or to ask your host/support how to get it.  Then you have simple labels without the need to repeat help text on every line.

brianteeman commented:
@mbabker except if you look at the text on the right for database host ;)

mbabker commented:
Didn't say that was perfect but it mostly illustrates what I would do without me having to invest time in clipping everything up to build a mock 😉

zwiastunsw commented:
Hmmm. There are 73 words in the Wordpress form. However, in the proposal for Joomla there are 66 words... And this text can be shortened to 48 more words.
* **Hostname** - "localhost" or a name provided by your host
* **Database type** - select from the list
* **Database Username** - own or set by your host
* **Database Password** - own or set by your host
* **Database Name** - own or set by your host
* **Table prefix** - own or generated by installer

But my English is weak.

C-Lodder commented:
@zwiastunsw - It's not about the overall word count, but about the word count in the label itself. I couldn't care less if we add a full paragraph of help text below each input as an explanation, but it's simply the current labels that are too long.

mbabker commented:
And I'm saying we don't need a help text for every field.  An introductory sentence, similar to the one WordPress uses, removes the mostly redundant "one you set up or were given" comment we have now on most every field.  Really we shouldn't need one for more than the hostname and table prefix with that.  Unless there is some major UX or a11y thing I'm missing that makes it more efficient to repeat that type of comment 5 or 6 times.

brianteeman commented:
@mbabker i agree with you - i was teasing about wp
It is more than good enough to have a heading text telling you where to get the information (either host or create it) and then just have labels

regarding word count. thats completely irrelevant. less words does not always mean it is easier to understand and it is also irrelevant for other languages. German users could probably use justasinglewordfortheentireinstallation

zwiastunsw commented:
I agree that the number of words is irrelevant.
We have two issues here:
* labels
* tints

Someone previously proposed to combine these elements into one. That is not a bad idea. But the consequence is longer texts (label + tips).
We can decide on one hint at the beginning of the form. That is also a good idea. 
Which solution is better? I do not know!  
Usability tests with beginning users may be the answer. This is a task for the UX team (@jonrz).
Our opinions should not be decisive because we know exactly what this is all about.

coolcat-creations commented:
I just wonder again, why is that information even needed "own or set by your host" - could people try to type in the database from the pentagon or what do we prevent with this hint? Maybe I am missing something...

zwiastunsw commented:
Some hostings provide only one database, others need to create the database themselves.

infograf768 commented:
And we are still missing an "Advanced Users" button to be able to keep the same Database name and also the same table prefix when doing a new installation (and adding bak_ or not).
In fact it would not only concern "Advanced Users" as @zwiastunsw correctly remarks above as a user may have to use a unique database. Imagine the number of tables with different prefixes in that case if he/she wants to reinstall a site from scratch multiple times...

C-Lodder commented:
Ok I've changed the language strings around, added some helpers and tweaked the headers:

Can remove some of the form helpers if need be

@brianteeman @mbabker @coolcat-creations 

![screeny3](https://user-images.githubusercontent.com/2019801/33231043-2824b156-d1e6-11e7-88eb-bdbd7c49e9ce.png)

C-Lodder committed:
more updates

dgrammatiko commented:
@C-Lodder if @zwiastunsw is happy (after checking with screen reader) with your changes I'm also otherwise let's leave this as is for the time.

brianteeman commented:
Please change "Joomla name" to "site name"

dgrammatiko commented on installation/language/en-GB/en-GB.ini:
```diff
@@ -184,8 +188,10 @@ INSTL_GNU_GPL_LICENSE="GNU General Public License"
+INSTL_MCRYPT_SUPPORT_AVAILABLE="Mcrypt Support"
 INSTL_NOTICEMBLANGNOTDEFAULT="PHP mbstring language is not set to neutral. This can be set locally by entering <strong>php_value mbstring.language neutral</strong> in your <code>.htaccess</code> file."
 INSTL_NOTICEMBSTRINGOVERLOAD="PHP mbstring function overload is set. This can be turned off locally by entering <strong>php_value mbstring.func_overload 0</strong> in your <code>.htaccess</code> file."
+INSTL_NOTICEMCRYPTNOTAVAILABLE="Warning! PHP mcrypt extension should be installed or enabled. Without this, some features of Joomla will not be available."
```
Change `!` to something else

C-Lodder replied:
I just added this from the en-US language ad it was missing in en-GB

dgrammatiko replied:
comma will be better for screen readers,

brianteeman replied:
No thats not correct. it is perfectly correct to use the ! here

mbabker replied:
Umm, didn't we remove the mcrypt check completely anyway?

brianteeman commented:
And specify the database to connect Joomla to can simple be
Enter the database details

C-Lodder committed:
lang tweak

C-Lodder commented:
@brianteeman done

C-Lodder committed:
sitename

Quy commented:
`Specify the Host Name` - lowercase `Host Name`

`Host Name *` and `Database Name *` - lowercase `Name`?

Why say `Either` and `you created`? We don't do it with `Table prefix`.

How about?
`Specify a username or a username provided by your host.`

mbabker commented:
> I just wonder again, why is that information even needed "own or set by your host" - could people try to type in the database from the pentagon or what do we prevent with this hint? Maybe I am missing something...

@coolcat-creations It's to help guide users.  They may not even be aware they have to set up a database to run Joomla, or that these credentials are not the same as creating your user account.  For people who are running the installer consistently it's not text you really need, but think about this now in a new user perspective or a not-so-technical person's perspective.

zwiastunsw commented:
Unfortunately. For the user of the screen reader all the tips below the fields are useless. 
In order to use them, the user of the screen reader would need to constantly switch between browse mode (reading mode) and focus mode (the mode of interaction with the form).  The accessibility of the form has been significantly reduced.

brianteeman commented:
Not true. With the correct markup it is not a problem eg aria-described-by
See https://developer.paciellogroup.com/blog/2014/12/using-aria-describedby-to-provide-helpful-form-hints/ if you don't know how

zwiastunsw commented:
@brianteeman: Agrees. This could be done in this way.

brianteeman commented:
Which makes your comment pure FUD

dgrammatiko commented:
@brianteeman yes using `for` and an id for the `span` should be ok, but:
- that code is not here
- that code is requiring screen readers to comply to the ARIA-DESCRIBEDBY which might not always be the case. The reason to have descriptive labels in the first place was exactly that, not using aria attributes wherever possible.

Anyways this PR as is is breaking badly accessibility!

brianteeman commented:
aria-describedby is well supported
https://www.powermapper.com/tests/screen-readers/labelling/input-text-aria-describedby/
